src/coreclr/jit/codegenxarch.cpp

```diff
@@ -8625,7 +8625,7 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
                 continue;
             }
 
-            var_types storeType = varDsc->lvaArgType();
+            var_types storeType = varDsc->GetRegisterType();
```

Comment:
I assume we can do like we do in genJmpMethod:
```
        var_types storeType = varDsc->GetActualRegisterType(); // We own the memory and can use the full move.
```
could somebody confirm?
but if it goes to RC2 it is better to keep it conservative.